### PR TITLE
Add comment and release note item regarding recent CLI11 changes.

### PR DESCRIPTION
**Pull request by gunney1**
These minor follow-ups to https://github.com/LLNL/axom/pull/1401 were suggested in a recent team meeting.

1. Mention the change in the release notes
2. Add comment to note our changes, so we'd see it when upgrading CLI11.

### CODE REVIEW

**Review by rhornung67 — APPROVED** [review]
Thank you @gunney1

File: src/thirdparty/axom/CLI11.hpp [hunk 1]
```diff
@@ -552,7 +552,7 @@ namespace CLI {
 /// These codes are part of every error in CLI. They can be obtained from e using e.exit_code or as a quick shortcut,
 /// int values from e.get_error_code().
 enum class ExitCodes {
-    CLI11_Success = 0,
+  CLI11_Success = 0 /* Formerly Success, changed to work around X11 #defining Success */,
```
Comment (agcapps) [inline src/thirdparty/axom/CLI11.hpp:555]: Can we please scoot this back over two spaces?

Comment (agcapps) [reply src/thirdparty/axom/CLI11.hpp:555]: If the line is just too long, maybe take out the words "Formerly Success"

**Review by agcapps — APPROVED** [review]

File: src/thirdparty/axom/CLI11.hpp (hunk 1 above)
Comment (gunney1) [reply src/thirdparty/axom/CLI11.hpp:555]: Done.

**Pull request merged**